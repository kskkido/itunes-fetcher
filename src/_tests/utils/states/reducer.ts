import { Action } from 'redux';
import { rootReducer, RootState } from 'states/root/reducer';
import { noop } from './action';

// TODO fix types
export const batchedActionsReducer = (
  state: RootState | {},
  actions: Action[] = [noop]
) => (
  actions.reduce(
    rootReducer,
    state as RootState
  ) as RootState
);
