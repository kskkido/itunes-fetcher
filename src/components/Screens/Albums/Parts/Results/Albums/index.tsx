import React from 'react';
import Section from 'components/Shared/Section';
import ListColumn from 'components/Shared/ListColumn';
import Album from 'containers/Screens/Albums/Parts/Results/Album';

type AlbumsProps = {
  ids: string[];
};

const Albums: React.SFC<AlbumsProps> = ({
  ids,
}) => (
  <Section>
    <ListColumn>
      {ids.map(id => <Album key={id} id={id} />)}
    </ListColumn>
  </Section>
);

export default Albums;
