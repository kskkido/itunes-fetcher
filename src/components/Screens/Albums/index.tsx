import React from 'react';
import PageContainer from 'components/Shared/Layouts/PageContainer';
import ResultsAlbums from 'containers/Screens/Albums/Parts/Results/Albums';

const Albums: React.SFC = () => (
  <PageContainer>
    <ResultsAlbums />
  </PageContainer>
);

export default Albums;
