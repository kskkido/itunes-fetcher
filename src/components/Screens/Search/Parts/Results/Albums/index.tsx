import React from 'react';
import { NAVIGATION_SEARCH_ALBUMS } from 'routes/navigations';
import Link from 'components/Shared/Link';
import Heading from 'components/Shared/Heading';
import Section from 'components/Shared/Section';
import SectionHeader from 'components/Shared/SectionHeader';
import ListColumn from 'components/Shared/ListColumn';
import Album from 'containers/Screens/Search/Parts/Results/Album';

type AlbumsProps = {
  ids: string[];
};

const Albums: React.SFC<AlbumsProps> = ({
  ids,
}) => (
  <Section>
    <SectionHeader>
      <Heading level={3}>
        Albums
      </Heading>
      <Link to={NAVIGATION_SEARCH_ALBUMS.name}>
        See All...
      </Link>
    </SectionHeader>
    <ListColumn>
      {ids.map(id => <Album key={id} id={id} />)}
    </ListColumn>
  </Section>
);

export default Albums;
