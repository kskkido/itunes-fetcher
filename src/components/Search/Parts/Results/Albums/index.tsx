import React from 'react';
import { NAVIGATION_SEARCH_RESULTS_ALBUMS } from 'routes/navigations';
import Link from 'components/Shared/Link';
import Heading from 'components/Shared/Heading';
import Section from 'components/Search/Parts/Section';
import SectionHeader from 'components/Search/Parts/SectionHeader';
import ResultList from 'components/Search/Parts/ResultList';

type AlbumsProps = {
  ids: string[];
};

const Albums: React.SFC<AlbumsProps> = ({
  ids,
}) => (
  <Section>
    <SectionHeader>
      <Heading level={3}>
        Albums
      </Heading>
      <Link to={NAVIGATION_SEARCH_RESULTS_ALBUMS.name}>
        See All...
      </Link>
    </SectionHeader>
    <ResultList ids={ids} />
  </Section>
);

export default Albums;
