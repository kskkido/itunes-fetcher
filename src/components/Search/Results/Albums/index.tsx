import React from 'react';
import PageContainer from 'components/Shared/Layouts/PageContainer';
import ResultsAlbums from 'containers/Search/Results/Albums/Parts/Results/Albums';

const Albums: React.SFC = () => (
  <PageContainer>
    <ResultsAlbums />
  </PageContainer>
);

export default Albums;
