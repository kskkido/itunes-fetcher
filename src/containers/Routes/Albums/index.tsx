import { createStackNavigator } from 'react-navigation-stack';
import * as navigations from 'routes/navigations';
import Album from './Album/Root';

const Albums = createStackNavigator(
  {
    [navigations.NAVIGATION_ALBUMS_ALBUM.name]: {
      screen: Album,
    },
  },
  {
    initialRouteName: navigations.NAVIGATION_ALBUMS_ALBUM.name,
    headerMode: 'none',
  }
);

export default Albums;
