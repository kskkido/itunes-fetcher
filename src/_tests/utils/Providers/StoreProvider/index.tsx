import React from 'react';
import { Provider } from 'react-redux';
import { RootState } from 'states/root/reducer';
import { configureStore } from 'states';

type StoreProviderProps = {
  children: React.ReactNode;
  state: Partial<RootState>;
};

const StoreProvider: React.SFC<StoreProviderProps> = ({
  children,
  state,
}) => (
  <Provider store={configureStore(state)}>
    {children}
  </Provider>
);

export default StoreProvider;
