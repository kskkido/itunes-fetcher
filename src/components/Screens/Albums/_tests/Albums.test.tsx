/// <reference types="jest" />
import React from 'react';
import TestRenderer from 'react-test-renderer';
import StoreProvider from '_tests/utils/Providers/StoreProvider';
import { batchedActionsReducer } from '_tests/utils/states/reducer';
import Albums from '../';

const setup = () => ({
  tree: TestRenderer.create(
    <StoreProvider state={batchedActionsReducer({})}>
      <Albums />
    </StoreProvider>
  ),
});

describe('Albums', () => {
  it('should render as expected', () => {
    expect(setup().tree.toJSON()).toMatchSnapshot();
  });
});
