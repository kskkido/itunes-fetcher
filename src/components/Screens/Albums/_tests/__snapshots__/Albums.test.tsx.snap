// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums should render as expected 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View />
</RCTScrollView>
`;
